src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/PackageFeeds/ConsolidatePackageFeed.cs

```diff
@@ -55,7 +55,7 @@ public override async Task<SearchResult<IPackageSearchMetadata>> ContinueSearchA
             var packagesNeedingConsolidation = _installedPackages
                 .GroupById()
                 .Where(g => g.Count() > 1)
-                .Select(g => new PackageIdentity(g.Key, g.Max()))
+                .Select(g => new PackageIdentity(g.Key, g.OrderByDescending(x => x.Version).FirstOrDefault().Version))
```

Comment:
The FirstOrDefault here would throw if the collection "g" were empty. There is a where clause saying that g.Count() must be greater than 1 but then the FirstOrDefault doesn't make sense to use here. If a default were returned, trying to dereference it to access its Version property would throw.

Reply:
The Default will never happen, so the code is redundant, and can be written instead as:
```suggestion
                .Select(g => new PackageIdentity(g.Key, g.OrderByDescending(x => x.Version).First().Version))
```

Reply:
Fixed and added tests.